stetho/src/main/java/com/facebook/stetho/inspector/protocol/module/DOM.java

```diff
@@ -2,30 +2,315 @@
+    public String value;
+  }
+
+  private static class AttributeRemovedEvent {
+    @JsonProperty(required = true)
+    public int nodeId;
+
+    @JsonProperty(required = true)
+    public String name;
+  }
+
+  private static class ChildNodeInsertedEvent {
+    @JsonProperty(required = true)
+    public int parentNodeId;
+
+    @JsonProperty(required = true)
+    public int previousNodeId;
+
+    @JsonProperty(required = true)
+    public Node node;
+  }
+
+  private static class ChildNodeRemovedEvent {
+    @JsonProperty(required = true)
+    public int parentNodeId;
+
+    @JsonProperty(required = true)
+    public int nodeId;
+  }
+
+  @ChromeDevtoolsMethod
+  public void enable(JsonRpcPeer peer, JSONObject params) {
+    mPeerManager.addPeer(peer);
+  }
+
+  @ChromeDevtoolsMethod
+  public void disable(JsonRpcPeer peer, JSONObject params) {
+    mPeerManager.removePeer(peer);
   }
 
   @ChromeDevtoolsMethod
   public JsonRpcResult getDocument(JsonRpcPeer peer, JSONObject params) {
-    Node node = new Node();
-    node.nodeId = 1;
-    node.nodeType = 1;
-    node.nodeName = "";
-    node.nodeValue = "";
-    return node;
+    Object rootElement = mDOMProvider.getRootElement();
+    if (rootElement == null) {
+      return null;
+    }
+
+    Node rootNode = createNodeForElement(rootElement);
+
+    GetDocumentResponse result = new GetDocumentResponse();
+    result.root = rootNode;
+    return result;
+  }
+
+  @ChromeDevtoolsMethod
+  public void highlightNode(JsonRpcPeer peer, JSONObject params) {
+    HighlightNodeRequest request = mObjectMapper.convertValue(params, HighlightNodeRequest.class);
+    if (request.nodeId != null) {
+      Object element = getElementForNodeId(request.nodeId);
+      // TODO: pass in the HighlightConfig
+      mDOMProvider.highlightElement(element);
+    }
   }
 
   @ChromeDevtoolsMethod
   public void hideHighlight(JsonRpcPeer peer, JSONObject params) {
+    mDOMProvider.hideHighlight();
+  }
+
+  public static class HighlightNodeRequest {
+    @JsonProperty(required = true)
+    public HighlightConfig highlightConfig;
+
+    @JsonProperty(required = false)
+    public Integer nodeId;
+
+    @JsonProperty(required = false)
+    public String objectId;
+  }
+
+  public static class HighlightConfig {
+    @JsonProperty(required = false)
+    public Boolean showInfo;
+
+    @JsonProperty(required = false)
+    public RGBAColor contentColor;
+
+    @JsonProperty(required = false)
+    public RGBAColor paddingColor;
+
+    @JsonProperty(required = false)
+    public RGBAColor borderColor;
+
+    @JsonProperty(required = false)
+    public RGBAColor marginColor;
+  }
+
+  public static class RGBAColor {
+    @JsonProperty(required = true)
+    public int r;
+
+    @JsonProperty(required = true)
+    public int g;
+
+    @JsonProperty(required = true)
+    public int b;
+
+    @JsonProperty(required = false)
+    public Double a;
+  }
+
+  private void unhookElement(Object element) {
```

Comment:
I think this is fixed by moving the object id mapping into another class, but `unhook` isn't really a consistent verb here.  You're really mapping and unmapping objects.  Or adding and removing.  Not hooking and unhooking.